CHANGELOG.md

```diff
@@ -4,6 +4,11 @@
 
 Not yet released; provisionally v1.4.0 (may change).
 
+### Storage TX Interfaces
```

Comment:
Move this down to the "Storage" section?

Reply:
Is time ordered change easier? The change log hasn't been updated since the last release

Reply:
I don't know for sure, but looking at `CHANGELOG.md` as it is now, it feels like it's split by category rather than time, and the time is factored into versions.

Reply:
The topical ordering of the CHANGELOG file caused me to make [broken releases](https://github.com/google/trillian/releases/tag/v1.3.4) in the past, so I'm nervous about continuing the topical ordering scheme.

Reply:
I might be lacking some context (e.g. not sure how ordering in changelog can cause a broken release), but here is my mental model:

Topics in the changelog don't cover the entire history (it seems like your concern regards that). Topics are only for the stuff between the last release and `HEAD`. When a new release is created, this text is copied from changelog to the release notes, and in the changelog it is replaced by a very short version of it. The new topical list of changes is reset to empty.